src/linked_list.py

```diff
@@ -1,45 +1,45 @@
 # -*- coding: utf-8 -*-
-"""This file implements a basic linked list data structure."""
+"""File implements a basic linked list data structure."""
 from __future__ import unicode_literals
 
 
 class Node(object):
-    """This builds our Node class."""
+    """Building our Node class."""
 
     def __init__(self, data):
-        """Initial setup of Node objects."""
         self.data = data
         self.next_node = None
 
 
 class LinkedList(object):
-    """This builds LinkedList class."""
+    """Building our LinkedList class."""
 
     def __init__(self, data=None):
-        """Initial setup of LinkedList class."""
         self.head = None
-        try:
-            if data:
+        if data is not None:
+            try:
                 for item in data:
                     self.push(item)
-        except TypeError:
-            raise TypeError('Please enter an object that is iterable.')
+            except TypeError:
+                raise TypeError('Please enter an object that is iterable.')
 
     def push(self, data):
-        """Function builds push method of LinkedList class."""
+        """Insert data at the head of the list."""
         new_node = Node(data)
         new_node.next_node = self.head
         self.head = new_node
-        return self.head
 
     def pop(self):
-        """Function builds pop method of LinkedList class."""
+        """Remove data from the head of the list."""
         pop_node = self.head
-        self.head = self.head.next_node
-        return pop_node
+        try:
+            self.head = self.head.next_node
+        except AttributeError:
+            raise IndexError("You cannot pop from an empty list.")
```

Comment:
I appreciate the attempt at error handling, however remember that the default value of `Node.next_node` is `None`. Every node will have an attribute `.next_node`, so you would never raise an `AttributeError`. What you want to do is probably more along these lines:

```
if self.head.next_node is not None:
    reassign the head and return some value
else:
   raise that IndexError
```